tests/test_config.py

```diff
@@ -0,0 +1,1470 @@
+        get_openshift_session(conf, workflow.user_params['namespace'])
+
+    @pytest.mark.parametrize('config, valid', [
+        ("""\
+operator_manifests:
+  allowed_registries: null
+        """, True),  # minimal valid example, allows all registries
+        ("""\
+operator_manifests:
+  allowed_registries:
+    - foo
+    - bar
+  repo_replacements:
+    - registry: foo
+      package_mappings_url: https://somewhere.net/mapping.yaml
+  registry_post_replace:
+    - old: foo
+      new: bar
+        """, True),  # all known properties
+        ("""\
+operator_manifests: null
+        """, False),  # has to be a dict
+        ("""\
+operator_manifests: {}
+        """, False),  # allowed_registries is required
+        ("""\
+operator_manifests:
+  allowed_registries: []
+        """, False),  # if not null, allowed_registries must not be empty
+        ("""\
+operator_manifests:
+  allowed_registries: null
+  something_else: null
+        """, False),  # additional properties not allowed
+        ("""\
+operator_manifests:
+  allowed_registries: null
+  registry_post_replace:
+    - old: foo
+        """, False),  # missing replacement registry
+        ("""\
+operator_manifests:
+  allowed_registries: null
+  registry_post_replace:
+    - new: foo
+        """, False),  # missing original registry
+        ("""\
+operator_manifests:
+  allowed_registries: null
+  repo_replacements:
+    - registry: foo
+        """, False),  # missing package mappings url
+        ("""\
+operator_manifests:
+  allowed_registries: null
+  repo_replacements:
+    - package_mappings_url: https://somewhere.net/mapping.yaml
+        """, False),  # missing registry
+        ("""\
+operator_manifests:
+  allowed_registries: null,
+  repo_replacements:
+    - registry: foo
+      package_mappings_url: mapping.yaml
+        """, False),  # package mappings url is not a url
+    ])
+    def test_get_operator_manifests(self, tmpdir, config, valid):
+        config += "\n" + REQUIRED_CONFIG
+        if valid:
+            read_yaml(config, 'schemas/config.json')
+        else:
+            with pytest.raises(OsbsValidationException):
+                read_yaml(config, 'schemas/config.json')
+            return
+
+        filename = os.path.join(str(tmpdir), 'config.yaml')
+        with open(filename, 'w') as fp:
+            fp.write(dedent(config))
+        conf = Configuration(config_path=str(tmpdir))
+
+        operator_config = conf.operator_manifests
+        assert isinstance(operator_config, dict)
+        assert "allowed_registries" in operator_config
+
+    def test_set_user_params(self, tmpdir):
+        filename = os.path.join(str(tmpdir), 'config.yaml')
+        with open(filename, 'w') as fp:
+            fp.write(dedent(REQUIRED_CONFIG))
+
+        workflow = self.prepare()
+        Configuration(config_path=str(tmpdir))
+
+        for k, v in USER_PARAMS.items():
+            assert workflow.user_params[k] == v
```

Comment:
This doesn't test anything related to configuration, does it?

Reply:
right moving to inner